pepper-apis/src/main/java/org/broadinstitute/ddp/util/SendGridMailUtil.java

```diff
@@ -90,4 +96,21 @@ private static void submitMailMessage(Mail mailMessage, String sendGridApiKey) {
             throw new RuntimeException(msg);
         }
     }
+
+    private static SendGrid createSendGridInstance(String sendGridApiKey, String proxy) {
+        var httpClientBuilder = HttpClients.custom();
+        if (proxy != null && !proxy.isBlank()) {
+            URL proxyUrl;
+            try {
+                proxyUrl = new URL(proxy);
+            } catch (MalformedURLException e) {
+                throw new IllegalArgumentException("proxy needs to be a valid url");
+            }
+            httpClientBuilder.setProxy(new HttpHost(proxyUrl.getHost(), proxyUrl.getPort(), proxyUrl.getProtocol()));
+            httpClientBuilder.setConnectionReuseStrategy(NoConnectionReuseStrategy.INSTANCE);
+            LOG.info("Using SendGrid proxy: {}", proxy);
+        }
+        var client = new Client(httpClientBuilder.build());
+        return new SendGrid(sendGridApiKey, client);
```

Comment:
Can we add a new public method in `SendGridClient` that returns the raw sendgrid object
```
public SendGrid getApiInterface() {
  return this.sendGrid;
}
```
and use that here instead of duplicating the code?